### PR TITLE
Statistics subsystem: new available statistics, plotting scripts, tests

### PR DESCRIPTION
**Pull request by Piccions**
This PR aims at improving the usability and the capabilities of the statistics subsystem used to take internal measurements of the simulator behaviour. In particular:
* The following statistics have been inserted in the simulator, the python interface and the default generated textual report:
    - Anti-messages count
    - Recovery costs, intended as the cost to restore a LP checkpoint and send the related anti-messages; the coasting forward operation costs are captured in another statistics value
    - Size of LP states when checkpointed
    - Costs of checkpoints
    - Event execution costs
    - Count of LPs
* A comprehensive test for the statistics subsystem, including the related python scripts, has been implemented
* Several bug related to the statistics subsystem have been fixed
    - Serial runtime statistics issues
    - Edge cases with divisions by zero in absence of rollbacks/checkpoints/processes messages

### CODE REVIEW

**Comment by Piccions** [comment]
This PR is a work in progress, things are missing and/or broken (we are still missing the full suite of python plotting scripts :cry: ).
I would appreciate some opinions on how to properly test the statistics python scripts. Right now the approach is kind of hacky: I invoke the python interpreter from C code using the ```system()``` function and the resulting textual report is checked with the help of another python script. Yes, also this latter script is invoked using ```system()```. Any good ideas in this regard?

**Comment by alessandropellegrini** [comment]
Another possibility can be to have python tests, not necessarily using a python testing framework (if the number of python tests increases it can be introduced).

We can launch these python tests directly from ctest:

```
find_package(Python REQUIRED)
add_test(python_test_whateber
   ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/tests/log/test.py
  )
```

From there, we could import the various functions from the scripts and test them. 
The binary file could be an asset in the test directory.

This does not test the generation-parsing interaction. In that case we could call the simple model from python (using some `$<TARGET_FILE:model>` generator).

**Comment by Piccions** [comment]
Unfortunately ```FindPython``` is not available in CMake 3.10, it has been introduced in the 3.12 version. We can still write a Python test though, but we will have to hardcode the Python interpreter path.
Also, I feel like the binary file shouldn't be a static asset in the repository since the test wouldn't break when it should, e.g.: after heavy changes in the statistics module.
So my new proposal would be to split the current test in two parts: the one that tests the C code and the one that tests the Python parsing script. We can then impose a dependency relationship on those two tests (the Python parsing script consumes the output of the C test). CMake offers [this feature](https://cmake.org/cmake/help/latest/prop_test/FIXTURES_REQUIRED.html) to do this!

**Comment by alessandropellegrini** [comment]
I don't see any major problem in bumping the CMake version to 3.12 though, because we are anyhow planning to package everything on our side. In this case, having a "newer" dependency is less critical if we can directly provide binaries.

Regarding the test dependency, that's the solution I like the most. I couldn't check this feature's availability (again, luckily, I'm not the CMake expert! :smile:).

**Comment by Piccions** [comment]
The only issue in bumping the version to 3.12 is that Ubuntu 18 LTS ships with CMake version 3.10. Yeah, not a big issue. If you agree, we will bump CMake version to 3.12!

**Comment by alessandropellegrini** [comment]
I see. Maintenance updates should be already ended though, but security support should last for 4 additional years. We're using a later version in the CI anyhow, right?
I think that it makes sense to cut an LTS that's approaching to some extent its EOL to have a cleaner code base.

**Review by alessandropellegrini — APPROVED** [review]
Thanks, looks good! Just one minor comment in the review.

File: src/gvt/termination.c [hunk 1]
```diff
@@ -84,12 +84,15 @@ void termination_on_gvt(simtime_t current_gvt)
...
  * @brief Force termination of the simulation
- *
- * FIXME this doesn't actually work: concurrent termination messages will break this
  */
-void termination_force(void)
+void RootsimStop(void)
 {
-	nid_t i = atomic_load_explicit(&nodes_to_end, memory_order_relaxed);
+	if(global_config.serial) {
+		global_config.termination_time = -1.0;
```
[the remark this answers is not in the document]
Comment (Piccions) [reply src/gvt/termination.c:91]: Yep! The same code path for regular simulation termination is followed; statistics are normally dumped

**Pull request merged**